apps-integration-tests/integration-tests-data-index-service/integration-tests-data-index-service-common/src/main/java/org/kie/kogito/index/AbstractProcessDataIndexIT.java

```diff
@@ -209,6 +216,65 @@ public void testProcessInstanceEvents() {
                             .body("data.Approvals[0].metadata.userTasks", is(notNullValue()))
                             .body("data.Approvals[0].metadata.userTasks.size()", is(2)));
         }
+
+        String pId2 = createTestProcessInstance();
+        await()
+                .atMost(TIMEOUT)
+                .untilAsserted(() -> getProcessInstanceById(pId2, "ACTIVE"));
+
+        await()
+                .atMost(TIMEOUT)
+                .untilAsserted(() -> given().spec(dataIndexSpec()).contentType(ContentType.JSON)
+                        .body("{ \"query\" : \"query { ProcessInstances (where: { id: {equal: \\\"" + pId2 + "\\\"}}) {nodeDefinitions {id}} }\" }")
+                        .when().post("/graphql")
+                        .then()
+                        .statusCode(200)
+                        .body("data.ProcessInstances.nodeDefinitions.size()", is(1)));
+
+        await()
+                .atMost(TIMEOUT)
+                .untilAsserted(() -> given().spec(dataIndexSpec()).contentType(ContentType.JSON)
+                        .body("{ \"query\" : \"mutation{ ProcessInstanceAbort( id: \\\"" + pId2 + "\\\")}\"}")
```

Comment:
would this work? I feel that query should not be used in this case
```suggestion
                        .body("{ \"mutation\" : \"ProcessInstanceAbort( id: \\\"" + pId2 + "\\\")}\"}")
```

Reply:
AFAIK this only work if we pass the operation in field named 'query'.. https://graphql.org/learn/serving-over-http/#post-request